apps/studio/src/App.vue

```diff
@@ -42,194 +33,538 @@
+        await this.$nextTick();
+        this.validateLicenseExpiry();
       }
     },
   },
   async beforeDestroy() {
-    clearInterval(this.interval)
-    clearInterval(this.licenseInterval)
+    clearInterval(this.interval);
+    clearInterval(this.licenseInterval);
   },
   async mounted() {
-    this.notifyFreeTrial()
-    this.interval = setInterval(this.notifyFreeTrial, globals.trialNotificationInterval)
-    this.$store.dispatch('licenses/updateAll');
-    this.licenseInterval = setInterval(
-      () => this.$store.dispatch('licenses/updateAll'),
-      globals.licenseCheckInterval
-    )
-    const query = querystring.parse(window.location.search, { parseBooleans: true })
-    if (query) {
-      this.url = query.url || null
-      this.runningWayland = !!query.runningWayland
-    }
+    try {
+      this.notifyFreeTrial();
+      this.interval = setInterval(
+        this.notifyFreeTrial,
+        globals.trialNotificationInterval
+      );
+      this.$store.dispatch("licenses/updateAll");
+      this.licenseInterval = setInterval(
+        () => this.$store.dispatch("licenses/updateAll"),
+        globals.licenseCheckInterval
+      );
 
+      // Initialize settings
+      await this.$store.dispatch("settings/initializeSettings");
 
-    this.$nextTick(() => {
-      window.main.isReady();
-    })
-    if (this.themeValue) {
-      document.body.className = `theme-${this.themeValue}`
-    }
+      // Ensure themes are loaded
+      await this.$store.dispatch("themes/fetchThemes");
+
+      // Initialize themes (ensure this happens after settings are loaded)
+      if (this.themeValue) {
+        console.log(`Initializing theme: ${this.themeValue}`);
+
+        // Apply the theme directly
+        if (typeof this.themeValue === "string") {
+          this.applyThemeCSS(this.themeValue);
+        } else if (this.themeValue.themeId) {
+          this.applyThemeCSS(
+            this.themeValue.themeId,
+            this.themeValue.css,
+            this.themeValue.baseTheme
+          );
+        }
+      }
+
+      // Listen for theme preview changes from ThemeManagerModal
+      this.$root.$on("theme-preview-changed", (payload) => {
+        console.log(
+          "Theme preview changed event received in App.vue:",
+          payload
+        );
+
+        if (typeof payload === "string") {
+          // Handle legacy format (just themeId)
+          this.applyThemeCSS(payload);
+        } else if (payload && payload.themeId) {
+          // Handle new format with CSS content
+          this.applyThemeCSS(payload.themeId, payload.css, payload.baseTheme);
+        }
+      });
 
-    if (this.url) {
-      try {
-        await this.$store.dispatch('openUrl', this.url)
-      } catch (error) {
-        console.error(error)
-        this.$noty.error(`Error opening ${this.url}: ${error}`)
-        throw error
+      // Also listen for window events
+      window.addEventListener("theme-preview", (event) => {
```

Comment:
You should use our Vue event bus for this - $this.$root.emit, and `registerHandlers`, along with `AppEvent`

Reply:
ok cool, will get that sorted out

Reply:
updated